lib/mixlib/config.rb

```diff
@@ -53,7 +53,40 @@ def initialize_mixlib_config
     # === Parameters
     # filename<String>:: A filename to read from
     def from_file(filename)
-      instance_eval(IO.read(filename), filename, 1)
+      if %w{ .yml .yaml }.include?(File.extname(filename))
+        instance_eval(from_yaml(filename), filename, 1)
+      else
+        instance_eval(IO.read(filename), filename, 1)
+      end
+    end
+
+    # Parses valid YAML structure into Ruby so it can be ingested into the Class
+    #
+    # === Parameters
+    # filename<String>:: A filename to read from
+    def from_yaml(filename)
+      require "yaml"
+      from_hash(YAML.load(IO.read(filename)))
+    end
+
+    # Transforms a Hash into method-style configuration syntax to be processed
+    #
+    # === Parameters
+    # hash<Hash>:: A Hash containing configuration
+    def from_hash(hash)
+      ruby_translation = []
+
+      to_dotted_hash(hash).each do |k, v|
+        if v.is_a? Array
+          ruby_translation << "#{k} #{v}"
+        elsif v.is_a? String
+          ruby_translation << "#{k} \"#{v}\""
+        else
+          ruby_translation << "#{k} #{v}"
+        end
+      end
+
+      ruby_translation.join("\n")
```

Comment:
i would expect these APIs to do the `instance_eval` here as well so that `Config.from_hash(hash)` and `Config.from_yaml(yaml)` would work stand-alone.

Reply:
well not "as well" but to move the `instance_eval()`s down into these methods....

Reply:
or just here since from_yaml calls this...  eventually the coffee will hit my brain...